src/browser/api_protocol/api_handlers/authorization.js

```diff
@@ -211,6 +211,26 @@ function AuthorizationApiHandler() {
 
     /*jshint unused:false */
     apiProtocolBase.registerActionMap(authenticationApiMap);
-}
+};
+
+const isConnectionAuthenticated = (msg, next) => {
+    const { data, nack, identity, strategyName } = msg;
+    const { uuid } = identity;
+    const action = data && data.action;
+
+    // Prevent all API calls from unauthenticated external connections,
+    // except for authentication APIs
+    if (
```

Comment:
Done it this way initially to short-circuit checks, but agree with _clarity over cleverness_.